### ISSUE TITLE
onselect gives setImmediate is not defined

### ISSUE THREAD

**Issue by utpalkantsharma** [oferh/ng2-completer#187]
I am using the same example what you have shared. Here is few code which i am using in my project

```
 <span>Search Captains</span>
   <ng2-completer [(ngModel)]="captain" [datasource]="captains" [minSearchLength]="0" (selected)="onSelected($event)"></ng2-completer>
```
I am getting list of captains as auto-complete, but onselect of option it is throwing an error setImmediate is not defined.
![screenshot from 2017-04-04 12 33 08](https://cloud.githubusercontent.com/assets/8805987/24645044/2073f496-1934-11e7-984a-02576c172312.png)

**Comment by oferh** [comment]
Hi,
can you provide a plunk?

**Comment by utpalkantsharma** [comment]
I created plunk and found it working perfectly fine (basically could not reproduce it in plunk).
I think angular version might be problem in my local system, refer image for error while installing
![screenshot from 2017-04-04 15 50 55](https://cloud.githubusercontent.com/assets/8805987/24652890/2b76a614-1951-11e7-9e04-e2bfb168c311.png)

**Comment by duffers20** [comment]
Hi oferh, I have also been experiencing this issue. In order to help with the debugging, I have created a project which can be cloned to recreate the issue (see Readme.md) for full details on reproducing the error:

https://github.com/duffers20/ng2-completer-issue-demo

**Comment by brandhaug** [comment]
+1

The error only occurs when selecting an item from the dropdown with a mouse click. Works fine when using keys to select.

**Comment by oferh** [comment]
closed in 1.3.1
